x/datapool/keeper/pool.go

```diff
@@ -63,3 +68,98 @@ func (k Keeper) SetDataValidator(ctx sdk.Context, dataValidator types.DataValida
 	store.Set(dataValidatorKey, bz)
 	return nil
 }
+
+func (k Keeper) IsRegisteredDataValidator(ctx sdk.Context, dataValidatorAddress sdk.AccAddress) bool {
+	store := ctx.KVStore(k.storeKey)
+	dataValidatorKey := types.GetKeyPrefixDataValidator(dataValidatorAddress)
+	return store.Has(dataValidatorKey)
+}
+
+func (k Keeper) CreatePool(ctx sdk.Context, curator sdk.AccAddress, poolParams types.PoolParams) (uint64, error) {
+	// Get the next pool id
+	poolID := k.GetNextPoolNumberAndIncrement(ctx)
+
+	// new pool
+	newPool := types.NewPool(poolID, curator, poolParams)
+
+	// pool address for deposit
+	poolAddress, err := types.AccPoolAddressFromBech32(newPool.GetPoolAddress())
+	if err != nil {
+		return 0, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address of pool %s", newPool.GetPoolAddress())
+	}
+
+	// check if account with pool address exists or not
+	acc := k.accountKeeper.GetAccount(ctx, poolAddress)
+	if acc != nil {
+		return 0, sdkerrors.Wrapf(types.ErrPoolAlreadyExist, "pool %d already exist", poolID)
+	}
```

Comment:
Did you check this because of duplicate poolIDs?

Reply:
You are right. To be precise, it checks if the account according to the address generated using poolID exists or not. Since new account under the address is generated when creating pool.

Reply:
I got your point. Yes. If collision occurs when creating pool address(it is generated by hashing), it will fail to create pool and will continue forever. 

We use hash for generating pool address assuming there is no collision. So I think there is no need to check if the address exists or not under this assumption (If collision occurs, we need a workaround for hash collision as you said, and it is not the only problem of this part. e.g. datadeal). Anyway I'll remove this check since there is no meaning under this assumption.

Reply:
Yes, That will be fine. 😀

Reply:
removed in [5f69ad0](https://github.com/medibloc/panacea-core/pull/287/commits/5f69ad06762947b77e90421a4b2487f058772fa2)